### PR TITLE
build(android_intent_plus): Update to target and compile SDK 34 on Android

### PR DESCRIPTION
**Pull request by vbuberen**
## Description

Same as #2701 but for `android_intent_plus`. 

Additionally, fixed the example app as it didn't even build locally.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate that with a `!` in the title as explained in [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0)).
- [x] No, this is *not* a breaking change.

### CODE REVIEW

File: packages/android_intent_plus/android/build.gradle [hunk 1]
```diff
@@ -46,7 +46,8 @@ android {
 }
 
 dependencies {
-    implementation 'androidx.annotation:annotation:1.7.0'
+    implementation(platform("org.jetbrains.kotlin:kotlin-bom:1.8.10"))
```
Comment (vbuberen) [inline packages/android_intent_plus/android/build.gradle:49]: This is one of ways to resolve build errors due to different AndroidX packages having different Kotlin versions.

See https://youtrack.jetbrains.com/issue/KT-55297/kotlin-stdlib-should-declare-constraints-on-kotlin-stdlib-jdk8-and-kotlin-stdlib-jdk7

File: packages/android_intent_plus/example/android/app/src/main/java/io/flutter/plugins/androidintentexample/MainActivity.java [hunk 2]
```diff
@@ -1,17 +1,22 @@
...
+import androidx.core.content.ContextCompat;
+
 import io.flutter.embedding.android.FlutterActivity;
 
 public class MainActivity extends FlutterActivity {
 
-  @Override
-  protected void onCreate(Bundle savedInstanceState) {
-    super.onCreate(savedInstanceState);
+    @SuppressLint("WrongConstant")
```
Comment (vbuberen) [inline packages/android_intent_plus/example/android/app/src/main/java/io/flutter/plugins/androidintentexample/MainActivity.java:13]: Here is the same thing: https://github.com/fluttercommunity/plus_plugins/pull/1811/files#r1193861724

Looks like Google folks don't really care about fixing this bug.

File: packages/android_intent_plus/android/build.gradle (hunk 1 above)
Comment (vbuberen) [reply packages/android_intent_plus/android/build.gradle]: That fix from above didn't work on CI, so replaced with another one, which I use in other projects: https://youtrack.jetbrains.com/issue/KT-54136/Duplicated-classes-cause-build-failure-if-a-dependency-to-kotlin-stdlib-specified-in-an-android-project#focus=Comments-27-6583109.0-0

**Comment by vbuberen** [comment]
It is so strange that on CI build fails https://github.com/fluttercommunity/plus_plugins/actions/runs/8308234561/job/22738260821?pr=2711

While locally for me both options on declaring constraints for Kotlin version work fine.

**Review by miquelbeltran — APPROVED** [review]

**Pull request merged**